blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/LocatorTest.java

```diff
@@ -0,0 +1,435 @@
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertEquals(0, locator.hashCode());
+    }
+
+    @Test
+    public void hashCodeReturnsHashCodeOfStringRepresentation() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey(fullyQualifiedName);
+
+        // expect
+        assertEquals(fullyQualifiedName.hashCode(), locator.hashCode());
+    }
+
+    @Test
+    public void equalsNullReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertFalse(locator.equals((Object)null));
+    }
+
+    @Test
+    public void equalsNonLocatorReturnsFalse() {
+
+        // given
+        Locator locator = new Locator();
+
+        // expect
+        assertFalse(locator.equals(new Object()));
+    }
+
+    @Test
+    public void equalsDifferentStringReturnsFalse() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = Locator.createLocatorFromDbKey("a.b.d");
+
+        // expect
+        assertFalse(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsSameStringReturnsTrue() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = Locator.createLocatorFromDbKey("a.b.c");
+
+        // expect
+        assertTrue(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsBothNullReturnsTrue() {
+
+        // given
+        Locator locator = new Locator();
+        Locator other = new Locator();
+
+        // expect
+        assertTrue(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsThisNullThrowsException() {
+
+        // given
+        Locator locator = new Locator();
+        Locator other = Locator.createLocatorFromDbKey("a.b.d");
+
+        // expect
+        assertFalse(locator.equals((Object)other));
+    }
+
+    @Test
+    public void equalsOtherNullThrowsException() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = new Locator();
+
+        // expect
+        assertFalse(locator.equals((Object)other));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void equals2ThisNullStringRepThrowsException() {
+
+        // given
+        Locator locator = new Locator();
+        Locator other = Locator.createLocatorFromDbKey("a.b.d");
+
+        // when
+        boolean result = locator.equals(other);
+
+        // then
+        // the exception is thrown
+    }
+
+    @Test
+    public void equals2OtherNullStringRepReturnsFalse() {
+
+        // given
+        Locator locator = Locator.createLocatorFromDbKey("a.b.c");
+        Locator other = new Locator();
+
+        // expect
+        assertFalse(locator.equals(other));
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void equals2OtherNullThrowsException() {
```

Comment:
I assume we'd rather not have these.  If so, would you make a comment that you are verifying current behavior and we'd rather return a false rather than null.